examples/multi-pointer.c

```diff
@@ -162,39 +128,186 @@ static void cursor_destroy(struct sample_cursor *cursor) {
 	free(cursor);
 }
 
-static void handle_input_remove(struct compositor_state *state,
-		struct wlr_input_device *device) {
-	struct sample_state *sample = state->data;
+void input_remove_notify(struct wl_listener *listener, void *data) {
+	struct wlr_input_device *device = data;
+	struct sample_cursor *sample_cursor = wl_container_of(listener, sample_cursor, destroy);
+	struct sample_state *sample = sample_cursor->sample;
 	struct sample_cursor *cursor;
 	wl_list_for_each(cursor, &sample->cursors, link) {
 		if (cursor->device == device) {
 			cursor_destroy(cursor);
 			break;
 		}
 	}
+	struct sample_pointer *pointer;
+	wl_list_for_each(pointer, &sample->pointers, link) {
+		if (pointer->device == device) {
+			free(pointer);
+			break;
+		}
+	}
+}
+
+void output_remove_notify(struct wl_listener *listener, void *data) {
+	struct sample_output *sample_output = wl_container_of(listener, sample_output, destroy);
+	struct sample_state *sample = sample_output->sample;
+	wl_list_remove(&sample_output->frame.link);
+	wl_list_remove(&sample_output->destroy.link);
+	wl_list_remove(&sample_output->link);
+	free(sample_output);
+
+	struct sample_cursor *cursor;
+	wl_list_for_each(cursor, &sample->cursors, link) {
+		configure_cursor(cursor->cursor, cursor->device, sample);
+	}
+}
+
+void new_output_notify(struct wl_listener *listener, void *data) {
+	struct wlr_output *output = data;
+	struct sample_state *sample = wl_container_of(listener, sample, new_output);
+	struct sample_output *sample_output = calloc(1, sizeof(struct sample_output));
+	if (wl_list_length(&output->modes) > 0) {
+		struct wlr_output_mode *mode = wl_container_of((&output->modes)->prev, mode, link);
```

Comment:
`(&output->modes)->prev` is maybe a Rust-ism? x)

Could be written as `output->modes.prev`